### PR TITLE
Collector internal telemetry updates

### PR DESCRIPTION
**Pull request by danelson**
- ~Internal collector detailed metrics were missing the `otelcol_` prefix.~
- `log_records` were missing from the critical monitoring sections (I assume this is not by design)

### CODE REVIEW

**Comment by danelson** [comment]
I just came across https://github.com/open-telemetry/opentelemetry-collector/issues/9315 and https://github.com/open-telemetry/opentelemetry-collector so now I am not sure this is correct. Perhaps it is because I am consuming the metrics on localhost:8888/metrics. Anyway, hopefully a maintainer can clarify.

**Review by mx-psi — DISMISSED** [review]

File: content/en/docs/collector/internal-telemetry.md [hunk 1]
```diff
@@ -237,28 +237,28 @@ categorized by instrumentation type.
...
+| `otelcol_rpc_client_response_size`        | Measures the size of RPC response messages (uncompressed).                                | Histogram |
+| `otelcol_rpc_client_responses_per_rpc`    | Measures the number of messages sent per RPC. Should be 1 for all non-streaming RPCs.     | Histogram |
+| `otelcol_rpc_server_duration`             | Measures the duration of inbound RPC.                                                     | Histogram |
+| `otelcol_rpc_server_request_size`         | Measures the size of RPC request messages (uncompressed).                                 | Histogram |
+| `otelcol_rpc_server_requests_per_rpc`     | Measures the number of messages received per RPC. Should be 1 for all non-streaming RPCs. | Histogram |
+| `otelcol_rpc_server_response_size`        | Measures the size of RPC response messages (uncompressed).                                | Histogram |
+| `otelcol_rpc_server_responses_per_rpc`    | Measures the number of messages sent per RPC. Should be 1 for all non-streaming RPCs.     | Histogram |
```
Comment (codeboten) [inline content/en/docs/collector/internal-telemetry.md:261]: These metrics are generated by the underlying instrumentation library, not by collector components. This means that the prefix will not be present here

**Review by codeboten — COMMENTED** [review]
Thanks @danelson, it's a bit confusing, but these metrics listed in the table will not be prefixed because they're emitted by instrumentation libraries rather than by collector components

File: content/en/docs/collector/internal-telemetry.md [hunk 2]
```diff
@@ -317,19 +318,21 @@ logs for messages such as `Dropping data because sending_queue is full`.
...
-returned to clients. Depending on the deployment and the clients' resilience,
-this might indicate clients' data loss.
+Sustained rates of `otelcol_receiver_refused_log_records`,
+`otelcol_receiver_refused_spans`, and `otelcol_receiver_refused_metric_points`
+indicate that too many errors were returned to clients. Depending on the
+deployment and the clients' resilience, this might indicate clients' data loss.
 
-Sustained rates of `otelcol_exporter_send_failed_spans` and
+Sustained rates of `otelcol_exporter_send_failed_log_records`,
+`otelcol_exporter_send_failed_spans` and
```
Comment (tiffany76) [inline content/en/docs/collector/internal-telemetry.md:327]:
```suggestion
`otelcol_exporter_send_failed_spans`, and
```

File: content/en/docs/collector/internal-telemetry.md [hunk 3]
```diff
@@ -317,19 +318,21 @@ logs for messages such as `Dropping data because sending_queue is full`.
...
 with the network or backend receiving the data.
 
 #### Data flow
 
-You can monitor data ingress with the `otelcol_receiver_accepted_spans` and
-`otelcol_receiver_accepted_metric_points` metrics and data egress with the
+You can monitor data ingress with the `otelcol_receiver_accepted_log_records`,
+`otelcol_receiver_accepted_spans`, and `otelcol_receiver_accepted_metric_points`
+metrics and data egress with the `otelcol_exporter_sent_log_records`,
 `otelcol_exporter_sent_spans` and `otelcol_exporter_sent_metric_points` metrics.
```
Comment (tiffany76) [inline content/en/docs/collector/internal-telemetry.md:338]:
```suggestion
metrics and data egress with the `otelcol_exporter_sent_log_records`,
`otelcol_exporter_sent_spans`, and `otelcol_exporter_sent_metric_points` metrics.
```

**Review by tiffany76 — CHANGES_REQUESTED** [review]
Two small changes, along with Alex's note. Thanks!

**Comment by svrnm** [comment]
@danelson please take a look at the requested changes, thanks!

File: content/en/docs/collector/internal-telemetry.md (hunk 1 above)
Comment (danelson) [reply content/en/docs/collector/internal-telemetry.md:261]: @codeboten are you saying to rever this? Also can you clarify how this affects consuming the data when scraping the metrics with the prometheus receiver? I am running 0.104.0 with the below config and I see metrics with names like `otelcol_http_server_response_size` (note it is not `response_body_size` either)
image

<details>
<summary>OTel config</summary>

```yaml
receivers:
  otlp:
    protocols:
      http:
        endpoint: 0.0.0.0:4318
  prometheus/collector:
    config:
      scrape_configs:
        - job_name: "internal"
          scrape_interval: 10s
          static_configs:
            - targets:
                - "localhost:8888"

processors:
  filter/collector:
    error_mode: ignore
    metrics:
      include:
        match_type: regexp
        metric_names:
          - .*http_server.*

exporters:
  debug:
    verbosity: detailed

service:
  telemetry:
    metrics:
      level: detailed
  pipelines:
    logs:
      receivers: [otlp]
      exporters: [debug]
    metrics:
      receivers: [prometheus/collector]
      processors: [filter/collector]
      exporters: [debug]
```
</details>

Comment (codeboten) [reply content/en/docs/collector/internal-telemetry.md:261]: Right, until v0.106.1, the prometheus exporter configured inside the collector was setting a prometheus namespace of `otelcol` to prefix all metrics exported via prometheus. This was inconsistent with the metrics exported via other exporters (OTLP, console). This was addressed by prefixing all collector component generated metrics manually with `otelcol_` to provide a consistent metric name for all exporters. Note that when I'm using the term exporters here, I mean the exporters configured inside the Collector for the use of the OTel Go SDK.

This means that all metrics generated by instrumentation libraries will match the names that these instrumentation libraries intended as per the example below, where `http_server_response_size` used to be prefixed by `otelcol_` and will now look like this:

```
http_server_response_size{http_method="POST",http_scheme="http",http_status_code="200",net_host_name="127.0.0.1",net_host_port="4318",net_protocol_name="http",net_protocol_version="1.1",service_instance_id="aa3d8988-fdf1-4023-8fff-193877983817",service_name="otelcontribcol",service_version="0.106.1-dev"} 2
```

Comment (danelson) [reply content/en/docs/collector/internal-telemetry.md]: Thank you. The disconnect I was having was between collector generated and instrumentation library generated metrics.

**Comment by danelson** [comment]
I think this should be good now. Thank you for the feedback.

**Review by tiffany76 — APPROVED** [review]

**Comment by tiffany76** [comment]
Thanks for the changes, @danelson! 

Also, I created #4933 to make sure we clarify the change in naming conventions with v0.106.1.

**Comment by cartermp** [comment]
@codeboten PTAL!

**Review by codeboten — APPROVED** [review]
Thanks @danelson!

**Review by cartermp — APPROVED** [review]

**Pull request merged**